baseapp/baseapp.go

```diff
@@ -165,13 +164,34 @@ func (app *BaseApp) LastBlockHeight() int64 {
 }
 
 // initializes the remaining logic from app.cms
-func (app *BaseApp) initFromStore(mainKey sdk.StoreKey) error {
+func (app *BaseApp) initFromMainStore(mainKey *sdk.KVStoreKey) error {
+
 	// main store should exist.
-	// TODO: we don't actually need the main store here
-	main := app.cms.GetKVStore(mainKey)
-	if main == nil {
+	mainStore := app.cms.GetKVStore(mainKey)
+	if mainStore == nil {
 		return errors.New("baseapp expects MultiStore with 'main' KVStore")
 	}
+
+	// memoize mainKey.
+	if app.mainKey != nil {
+		panic("app.mainKey expected to be nil; duplicate init?")
+	}
+	app.mainKey = mainKey
+
+	// load consensus param from the main store
+	consensusParamsBz := mainStore.Get(mainConsensusParamsKey)
+	if consensusParamsBz != nil {
+		var consensusParams = &abci.ConsensusParams{}
+		err := proto.Unmarshal(consensusParamsBz, consensusParams)
```

Comment:
We don't want to use a codec here?

Reply:
not sure @jaekwon

Reply:
switched to codec - Jae can switch back if there is a good reason. Otherwise yeah staying consistent makes sense

Reply:
It should be proto because abci.ConsensusParams are protobuf messages.